scootapi/server/serve.go

```diff
@@ -20,16 +23,70 @@ func Serve(handler scoot.Proc, addr string, transportFactory thrift.TTransportFa
 	return server.Serve()
 }
 
-type Handler struct{}
+type Handler struct {
+	queue queue.Queue
+}
 
-func NewHandler() scoot.Proc {
-	return &Handler{}
+func NewHandler(queue queue.Queue) scoot.Proc {
+	return &Handler{queue}
 }
 
-func (h *Handler) RunJob(def *scoot.JobDefinition) (*scoot.Job, error) {
+func (h *Handler) RunJob(def *scoot.JobDefinition) (*scoot.JobId, error) {
 	log.Println("Running job", def)
-	r := scoot.NewInvalidRequest()
-	msg := "Scoot is working by saying it won't work"
-	r.Message = &msg
-	return nil, r
+
+	job, err := thriftJobToScoot(def)
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := h.queue.Enqueue(job)
+
+	if err != nil {
+		switch err := err.(type) {
+		case *queue.InvalidJobRequest:
+			r := scoot.NewInvalidRequest()
+			r.Message = new(string)
+			*r.Message = err.Error()
+			return nil, r
+		case *queue.CanNotScheduleNow:
+			r := scoot.NewCanNotScheduleNow()
+			r.RetryAfterMs = new(int64)
+			*r.RetryAfterMs = int64(err.UntilRetry / time.Millisecond)
+			return nil, r
+		}
+		return nil, err
+	}
+
+	r := scoot.NewJobId()
+	r.ID = id
+	return r, nil
+}
+
+// Translates thrift job definition message to scoot domain object
+func thriftJobToScoot(def *scoot.JobDefinition) (result sched.Job, err error) {
```

Comment:
this method would be really good to use property based testing on.  To ensure a variety of supplied jobs get converted correctly.